src/common/Finisher.cc

```diff
@@ -50,11 +50,10 @@ void *Finisher::finisher_thread_entry()
     /// Every time we are woken up, we process the queue until it is empty.
     while (!finisher_queue.empty()) {
       // To reduce lock contention, we swap out the queue to process.
-      // This way other threads can submit new contexts to complete while we are working.
-      vector<Context*> ls;
-      list<pair<Context*,int> > ls_rval;
+      // This way other threads can submit new contexts to complete
+      // while we are working.
+      vector<pair<Context*,int>> ls;
```

Comment:
Do we need a pair here? Seems to be not handling anything other than a zero, if we are to handle the non zero values, then we can add it.

Reply:
other queue() method(s) can take a non-zero value of r